tests/Config/ConfigCreatorFacadeTest.php

```diff
@@ -0,0 +1,141 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Config;
+
+use Infection\Config\ConfigCreatorFacade;
+use Infection\Config\InfectionConfig;
+use Infection\Finder\Exception\LocatorException;
+use Infection\Finder\LocatorInterface;
+use Infection\Utils\TmpDirectoryCreator;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * @internal
+ */
+final class ConfigCreatorFacadeTest extends TestCase
+{
+    /**
+     * @var ConfigCreatorFacade
+     */
+    private $creatorFacade;
+
+    /**
+     * @var LocatorInterface|MockObject
+     */
+    private $locator;
+
+    /**
+     * @var Filesystem|MockObject
+     */
+    private $filesystemMock;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var string
+     */
+    private $workspace;
+
+    /**
+     * @var string
+     */
+    private $tmpDir;
+
+    protected function setUp(): void
+    {
+        $this->locator = $this->createMock(LocatorInterface::class);
+        $this->filesystemMock = $this->createMock(Filesystem::class);
+
+        $this->creatorFacade = new ConfigCreatorFacade($this->locator, $this->filesystemMock);
+
+        $this->filesystem = new Filesystem();
+        $this->workspace = sys_get_temp_dir() . \DIRECTORY_SEPARATOR . 'infection-test' . \microtime(true) . \random_int(100, 999);
+        $this->tmpDir = (new TmpDirectoryCreator($this->filesystem))->createAndGet($this->workspace);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->filesystem->remove($this->workspace);
+    }
+
+    public function test_it_creates_config_class_successfully(): void
+    {
+        $json = '{"timeout": 25, "source": {"directories": ["src"]}}';
+
+        $customConfigPath = $this->tmpDir . '/file.json';
+
+        $this->locator
+            ->method('locateOneOf')
+            ->willReturn($customConfigPath);
+
+        $this->filesystem->dumpFile($customConfigPath, $json);
+
+        $infectionConfig = $this->creatorFacade->createConfig($customConfigPath);
+
+        self::assertSame($infectionConfig->getProcessTimeout(), 25);
+    }
+
+    public function test_it_creates_empty_config_class_successfully(): void
```

Comment:
If we are testing the creation of an empty config class, why is  there json dumped into the config file?

Reply:
will change the names of these methods. Locator should return smth like it found some configuration file.